docs/proofs/example1.rst

```diff
@@ -0,0 +1,280 @@
+Example 1
+=========
+
+.. list-table:: Nat identity
+
+   * - As an example of elaborator reflection here we step through this identity implementation:
+     - Nat identity.
+
+       .. code-block:: idris
+
+         %language ElabReflection
+
+         idNat : Nat -> Nat
+         idNat = %runElab (do
+           intro `{{x}}
+           fill (Var `{{x}})
+           solve
+         )
+
+.. list-table:: Nat identity
+
+   * - We start off with a hole for the whole term (of type Nat->Nat).
+       We fill that with 'intro' which creates a lambda term with a hole for the  expression.
+     - diagram
+
+       .. image:: ../image/tree.png
+          :width: 133px
+          :height: 106px
+
+After each tactic we will look at the state:
+
+.. list-table:: evolution of state
+
+   * - We start with the type signature like this:
+     - Nat identity.
+
+       .. code-block:: idris
+
+         %language ElabReflection
+
+         idNat : Nat -> Nat
+         idNat = %runElab (do
+
+   * - In order to investigate how the program works we can look at the proofState at each stage as the tactics are applied. So here is the proofState at the start:
+     - state
+
+       .. image:: ../image/elabProofStateEx1_1.png
+          :width: 310px
+          :height: 124px
+
+   * - We can also look at the logic at each stage: 
+     - logic
+
+       .. image:: ../image/elabLogicEx1_1.png
+          :width: 126px
+          :height: 33px
+
+   * - The term is:
```

Comment:
Well, I wanted a way to see the exact contents and syntax of these results, I could not find a way to do this apart from writing this horrible hack here:

https://github.com/martinbaker/Idris-dev/blob/uglyTTPrinter/libs/prelude/Language/Reflection/TTPrinter.idr

What is the recommended way to see and display the full results from getEnv, getGoal, getHoles and getGuess? I am looking for something that can be used from within Idris (so it can be displayed by debugMessage or fail) not just for ElabReflection but also to help me get a more intuitive understanding of the syntax of TT and some of the internal workings.